contracts/nois-proxy/src/contract.rs

```diff
@@ -36,6 +36,7 @@ pub fn instantiate(
         withdrawal_address,
         test_mode,
         callback_gas_limit,
+        randomness_price,
```

Comment:
Here we have two variables 
- prices (how much a dapp has to pay the proxy) 
- and randomness_prices (how much the proxy will pay to the payment contract)
This might become confusing to people I think. Whether we need a better variable name/ better documentation, or make it one variable. and then for chains that want to allow payment with native coin we can add some optional price?

Reply:
maybe we could have an

```rust
enum OperationalMode {
    // Proxy not in charge of the randomness payment. payment contract has to be filled on Nois chain beforehand
    PrefillMode{ 
        withdrawal_address,
        callback_gas_limit,
        },
    // Proxy continously pays in ibc Nois token
    IBCTokenPay { 
        withdrawal_address,
        callback_gas_limit,
        randomness_price
        },

    // Native token pay
    NativePay {
        prices
        withdrawal_address,
        callback_gas_limit,
        randomness_price
        },
}
```

Reply:
Yeah this makes a lot of sense. Question is do we implement this as an enum or 3 different proxy contracts?

The last 2 modes would benefit from being able to query the price automatically. The first one does not care.

Reply:
I agree. The enum itself is no issue at all.

My thinking was more that if you have your own specialized proxy you want to strip off all the other logic. But well this is a luxury problem to have when people optimize their custom proxies.

Reply:
Which mode(s) should we implement now?

Reply:
I implemented `OperationalMode` now. This is how the proxy pays its bills and does not worry about how to proxy generates income.